crates/red_knot_test/src/lib.rs

```diff
@@ -221,6 +231,17 @@ fn run_test(
         }
     }
 
+    if let Some(python_path) = python_path {
+        if !has_custom_pyvenv_cfg_file {
+            let pyvenv_cfg_file = python_path.join("pyvenv.cfg");
+            let python_version = test.configuration().python_version().unwrap_or_default();
+            let home_directory = SystemPathBuf::from(format!("/Python{python_version}"));
+            db.create_directory_all(&home_directory).unwrap();
+            db.write_file(&pyvenv_cfg_file, format!("home = {home_directory}"))
+                .unwrap();
+        }
+    }
```

Comment:
Hmm, GitHub doesn't allow me to add a suggestion. 

Can you move the `let python_version = test.configuration().python_version().unwrap_or_default();` out of the `if let Some(python_path) = python_path` and use the same version in the `ProgramSettings` initializer. Just to ensure that we use the same version.